extensions/binding/org.eclipse.smarthome.binding.fsinternetradio/src/main/java/org/eclipse/smarthome/binding/fsinternetradio/handler/FSInternetRadioHandler.java

```diff
@@ -0,0 +1,232 @@
+        radioLogin(true);
+
+        // also schedule a thread for polling with configured refresh rate
+        final BigDecimal period = (BigDecimal) getThing().getConfiguration().get(CONFIG_PROPERTY_REFRESH);
+        if (period != null && period.intValue() > 0) {
+            updateJob = scheduler.scheduleAtFixedRate(updateRunnable, period.intValue(), period.intValue(),
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    private void radioLogin(boolean logFailure) {
+        // read configuration and spawn thread to establish connection to device
+        final String ip = (String) getThing().getConfiguration().get(CONFIG_PROPERTY_IP);
+        final BigDecimal port = (BigDecimal) getThing().getConfiguration().get(CONFIG_PROPERTY_PORT);
+        final String pin = (String) getThing().getConfiguration().get(CONFIG_PROPERTY_PIN);
+
+        if (ip == null || pin == null || port.intValue() == 0) {
+            // configuration incomplete
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Configuration incomplete");
+        } else {
+            new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    logger.debug("creating new connection to " + ip + ":" + port);
+                    try {
+                        final FrontierSiliconRadio tmpRadio = new FrontierSiliconRadio(ip, port.intValue(), pin);
+                        tmpRadio.login();
+                        radio = tmpRadio; // login successful, now store radio in field
+
+                        // Thing initialized. If done set status to ONLINE to indicate proper working.
+                        updateStatus(ThingStatus.ONLINE);
+
+                        // now update all channels!
+                        updateRunnable.run();
+                    } catch (Exception e) {
+                        String msg = "Could not establish connection to radio: " + e.getMessage();
+                        logger.error(msg);
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, msg);
+                    }
+                }
+            }).start();
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (updateJob != null)
+            updateJob.cancel(true);
+        updateJob = null;
+        radio = null;
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        if (radio == null) {
+            // connection to radio is not initialized, log ignored command and set status, if it is not already offline
+            logger.warn("Ignoring command " + channelUID.getId() + " = " + command + " because device is offline.");
+            if (ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Device seems to be offline.");
```

Comment:
"seems to" is a bit vague. I'd prefer to not add any detailed message if we do not exactly know a reason.